public/js/reducers/app.js

```diff
@@ -17,6 +19,28 @@ export default function app(state, action) {
       return Object.assign({}, state, {
         csrfToken: action.csrfToken,
       });
+    case actionTypes.ADD_NOTIFICATION:
+      var newNotifications = state.notifications.slice(0);
+      var notificationMap = new Map(newNotifications);
+      var notification = notificationMap.set(getId(), action.data);
+      newNotifications = [...notificationMap];
```

Comment:
huh, it's surprising to me that Map preserves order when deconstructed to an array but the docs say indeed it does.